src/tap/include/tap/tapcell.h

```diff
@@ -51,118 +51,136 @@ class dbBlock;
-           const char* tap_nwin3_master,
-           const char* tap_nwouttie_master,
-           const char* tap_nwout2_master,
-           const char* tap_nwout3_master,
-           const char* incnrcap_nwin_master,
-           const char* incnrcap_nwout_master,
-           const char* tapcell_master,
-           int& dist);
+           int halo_x,
+           int halo_y,
+           const std::string& cnrcap_nwin_master,
+           const std::string& cnrcap_nwout_master,
+           int add_boundary_cell,
+           const std::string& tap_nwintie_master,
+           const std::string& tap_nwin2_master,
+           const std::string& tap_nwin3_master,
+           const std::string& tap_nwouttie_master,
+           const std::string& tap_nwout2_master,
+           const std::string& tap_nwout3_master,
+           const std::string& incnrcap_nwin_master,
+           const std::string& incnrcap_nwout_master,
+           const std::string& tapcell_master,
+           int dist);
   void reset();
-  int makeSiteLoc(int x, double site_x, int dirc, int& offset);
+  int makeSiteLoc(int x, double site_x, bool at_left_from_macro, int offset);
   void buildRow(odb::dbBlock* block,
-                std::string name,
+                const std::string& name,
                 odb::dbSite* site,
                 int start_x,
                 int end_x,
                 int y,
                 odb::dbOrientType& orient,
                 odb::dbRowDir& direction,
-                int& min_row_width);
+                int min_row_width);
   void cutRows(odb::dbMaster* endcap_master,
-               std::vector<odb::dbBox*> blockages,
-               int& halo_x,
-               int& halo_y);
-  int removeCells(const char* prefix);
-  int insertEndcaps(std::vector<std::vector<odb::dbRow*>>& rows,
-                    odb::dbMaster* endcap_master,
-                    std::vector<std::string>& cnrcap_masters);
-  std::vector<std::vector<odb::dbRow*>> organizeRows();
-  int insertTapcells(std::vector<std::vector<odb::dbRow*>>& rows,
-                     std::string tapcell_master,
-                     int& dist);
-  int insertAtTopBottom(std::vector<std::vector<odb::dbRow*>>& rows,
-                        std::vector<std::string> masters,
-                        odb::dbMaster* endcap_master,
-                        std::string prefix);
-  int insertAroundMacros(std::vector<std::vector<odb::dbRow*>>& rows,
-                         std::vector<std::string>& masters,
-                         odb::dbMaster* corner_master,
-                         std::string prefix);
-  bool overlaps(odb::dbBox* blockage,
-                odb::dbRow* row,
-                int& halo_x,
-                int& halo_y);
+               const std::vector<odb::dbBox*>& blockages,
+               int halo_x,
+               int halo_y);
+  int removeCells(const std::string& prefix);
   std::vector<odb::dbBox*> findBlockages();
  
  private:
+  struct filled_sites
```

Comment:
Should be FilledSites (see https://google.github.io/styleguide/cppguide.html#Type_Names)